toolchain/driver/driver.cpp

```diff
@@ -789,7 +792,7 @@ class Driver::CompilationUnit {
            !input_filename_.starts_with(options_.exclude_dump_file_prefix);
   }
 
-  Driver* driver_;
+  DriverEnv& driver_env_;
```

Comment:
I would prefer fields not to have a reference type, particularly non-const. Can this be made into a pointer instead? I see that some other fields are references too -- what is the motivation?

Reply:
Generally I think there's a mix in the toolchain, tending more to references than pointers. One constraint of references is that they're implicitly non-null.

Reply:
Perhaps as a stronger argument, three lines down there's a &. This type should be self-consistent.

https://discord.com/channels/655572317891461132/821113559755784242/1283516297686286377 regarding this.

Reply:
Trying to read a possible conclusion from current discussion, updated to `DriverEnv*`